### PR TITLE
Nir/dee 482 create plot files for nlp drift

### PR DESCRIPTION
**Pull request by nirhutnik**
[Text Property Outliers — gallery-boilerplate documentation.pdf](https://github.com/deepchecks/deepchecks/files/11341222/Text.Property.Outliers.gallery-boilerplate.documentation.pdf)
[Prediction Drift — gallery-boilerplate documentation.pdf](https://github.com/deepchecks/deepchecks/files/11341223/Prediction.Drift.gallery-boilerplate.documentation.pdf)
[Label Drift — gallery-boilerplate documentation.pdf](https://github.com/deepchecks/deepchecks/files/11341224/Label.Drift.gallery-boilerplate.documentation.pdf)
[Property Label Correlation — gallery-boilerplate documentation.pdf](https://github.com/deepchecks/deepchecks/files/11341225/Property.Label.Correlation.gallery-boilerplate.documentation.pdf)

### CODE REVIEW

File: docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
+# -*- coding: utf-8 -*-
+"""
+.. _plot_nlp_property_label_correlation:
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py]:
```suggestion
.. _nlp__property_label_correlation:
```

File: deepchecks/vision/suites/default_suites.py [hunk 2]
```diff
@@ -48,7 +48,7 @@ def train_test_validation(label_properties: List[Dict[str, Any]] = None, image_p
              - :class:`~deepchecks.vision.checks.train_test_validation.ImagePropertyDrift`
            * - :ref:`plot_vision_image_dataset_drift`
              - :class:`~deepchecks.vision.checks.train_test_validation.ImageDatasetDrift`
-           * - :ref:`plot_vision_feature_label_correlation_change`
+           * - :ref:`plot_vision_property_label_correlation_change`
```
Comment (noamzbr) [inline deepchecks/vision/suites/default_suites.py:51]:
```suggestion
           * - :ref:`_nlp__property_label_correlation `
```

File: docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+What Is The Purpose of the Check?
+=================================
+The check estimates for every :doc:`text property </user-guide/nlp/nlp_properties>`
+(such as text length, language etc.) its ability to predict the label by itself.
+
+This check can help find a potential bias in the dataset - the labels being strongly correlated with simple text
+properties such as percentage of special characters, sentiment, toxicity and more.
+
+This is a critical problem, sometimes referred to as shortcut learning, where the model is likely to learn this property
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py:23]: Maybe start by mentioning shortcut learning? Better opener than "find a potential bias"

File: docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+instead of the actual linguistic characteristics of each class, as it's easier to do so. In this case, the model will
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py:24]:
```suggestion
instead of the actual textual content relating to each class, as it's easier to do so. In this case, the model will
```

File: docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+show high performance on text taken in similar conditions, but will fail in the wild, where the simple properties
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py:25]:
```suggestion
show high performance on text collected from a specific source, but will fail in the wild, where the simple properties
```

File: docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+don't hold true.
+This kind of correlation will likely stay hidden without this check until tested in the wild.
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py:27]: wild -> production?

File: docs/source/checks/nlp/data_integrity/plot_text_property_outliers.py [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+inspecting random samples. For example, it may help you understand you have some corrupt samples (e.g.
+texts without spaces between words), or samples you didn't expect to have (e.g. texts in Norwegian instead of English).
+In some cases, these outliers may help debug some performance discrepancies (the model can be excused for failing on
+a totally blank text). In more extreme cases, the outlier samples may indicate the presence of samples interfering with
+the model's training by teaching the model to fit "irrelevant" samples.
+
+
+How Does the Check Work?
+------------------------
+Ideally we would like to directly find text samples which are outliers, but this is computationally expensive and does not
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_text_property_outliers.py:31]: Don't know if we need all this explanation here. Can just say that this check is about samples that are outliers in the sense that they are extreme / anomalous in one of the pre-ordained set of properties.

File: docs/source/checks/nlp/data_integrity/plot_text_property_outliers.py [hunk 8]
```diff
@@ -0,0 +1,75 @@
...
+have a clear and explainable results. Therefore, we use text properties in order to find outliers (such as text length,
+average word length, language etc.) which are much more efficient to compute, and each outlier is easily explained.
+
+* For numeric properties (such as "percent of special characters"), we use
+  `Interquartile Range <https://en.wikipedia.org/wiki/Interquartile_range#Outliers>`_ to define our upper and lower
+  limit for the properties' values.
+* For categorical properties (such as "language"), we use our own method of "sharp drop" in category distribution to
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_text_property_outliers.py:38]:
```suggestion
* For categorical properties (such as "language"), we look for a "sharp drop" in the category distribution to
```

File: docs/source/checks/nlp/data_integrity/plot_text_property_outliers.py [hunk 9]
```diff
@@ -0,0 +1,75 @@
...
+dataset = tweet_emotion.load_data(as_train_test=False)
+
+check = TextPropertyOutliers()
+result = check.run(dataset)
+result
+
+#%%
+# Observe Graphic Result
+# ^^^^^^^^^^^^^^^^^^^^^^
+# In this example, we can find many tweets that are outliers - For example, in the "average word length" property,
```
Comment (noamzbr) [inline docs/source/checks/nlp/data_integrity/plot_text_property_outliers.py:71]: There are many plots here. Maybe show only a few plots in this example?

File: docs/source/checks/nlp/model_evaluation/plot_prediction_drift.py [hunk 10]
```diff
@@ -0,0 +1,105 @@
...
+Drift is simply a change in the distribution of data over time, and it is
+also one of the top reasons why machine learning model's performance degrades
+over time.
+
+Prediction drift is when drift occurs in the prediction itself.
+Calculating prediction drift is especially useful in cases
+in which labels are not available for the test dataset, and so a drift in the predictions
+is our only indication that a changed has happened in the data that actually affects model
+predictions. If labels are available, it's also recommended to run the
+:doc:`Label Drift check </checks_gallery/nlp/train_test_validation/plot_label_drift>`.
```
Comment (noamzbr) [inline docs/source/checks/nlp/model_evaluation/plot_prediction_drift.py:28]: This link is no longer correct in the new branch. Should be 
```suggestion
:doc:`Label Drift check </nlp/auto_checks/train_test_validation/plot_label_drift>`.
```

File: docs/source/checks/nlp/model_evaluation/plot_prediction_drift.py [hunk 11]
```diff
@@ -0,0 +1,105 @@
...
+
+result
+
+#%%
+# We can see that we found drift in the distribution of the predictions, and that the drift is mainly in the "anger"
+# class. This makes sense, as we dropped 50% of the "anger" tweets from the train dataset, and so the model is now
+# predicting less "anger" tweets in the test dataset.
+
+#%%
+# The prediction drift check can also calculate drift on each class separately rather than the probabilities.
```
Comment (noamzbr) [inline docs/source/checks/nlp/model_evaluation/plot_prediction_drift.py:89]:
```suggestion
# The prediction drift check can also calculate drift on the probability of each class separately rather than the final predicted class.
```

File: docs/source/checks/nlp/model_evaluation/plot_prediction_drift.py [hunk 12]
```diff
@@ -0,0 +1,105 @@
...
+train_probas = train_probas[indices_to_keep]  # Filter the probabilities again
+
+check = PredictionDrift(drift_mode='proba')
+result = check.run(train_dataset=train_ds, test_dataset=test_ds,
+                   train_probabilities=train_probas, test_probabilities=test_probas)
+
+result
+#%%
+# This time, we can see there's small drift in each class. The "anger" class drift is actually probably caused by low
+# sample size, and not by drift in the data itself, as we did not change the data within the class, but only changed
```
Comment (noamzbr) [inline docs/source/checks/nlp/model_evaluation/plot_prediction_drift.py:103]: how is low sample size causing this?

File: docs/source/checks/vision/data_integrity/plot_property_label_correlation.py [hunk 13]
```diff
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 """
-.. _plot_vision_feature_label_correlation:
+.. _plot_vision_property_label_correlation:
```
Comment (noamzbr) [inline docs/source/checks/vision/data_integrity/plot_property_label_correlation.py:3]: Shouldn't this be _vision_property... ? Can you compare to other vision checks in the new ver?

File: docs/source/checks/vision/train_test_validation/plot_property_label_correlation_change.py [hunk 14]
```diff
@@ -1,6 +1,6 @@
 # -*- coding: utf-8 -*-
 """
-.. _plot_vision_feature_label_correlation_change:
+.. _plot_vision_property_label_correlation_change:
```
Comment (noamzbr) [inline docs/source/checks/vision/train_test_validation/plot_property_label_correlation_change.py:3]: same here

File: docs/source/checks/nlp/data_integrity/plot_text_property_outliers.py (hunk 7 above)
Comment (nirhutnik) [reply docs/source/checks/nlp/data_integrity/plot_text_property_outliers.py:31]: I disagree... There's no other place where we can explain our logic, and I think this is actually important information

File: docs/source/checks/nlp/model_evaluation/plot_prediction_drift.py (hunk 12 above)
Comment (nirhutnik) [reply docs/source/checks/nlp/model_evaluation/plot_prediction_drift.py:103]: Low sample size creates a more noisy distribution.

File: docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py (hunk 3 above)
Comment (nirhutnik) [reply docs/source/checks/nlp/data_integrity/plot_property_label_correlation.py]: I think it's problematic to start from this as the "bias" line is required to explain shortcut learning.

File: docs/source/checks/vision/data_integrity/plot_property_label_correlation.py (hunk 13 above)
Comment (nirhutnik) [reply docs/source/checks/vision/data_integrity/plot_property_label_correlation.py:3]: Fixed

File: docs/source/checks/vision/train_test_validation/plot_property_label_correlation_change.py (hunk 14 above)
Comment (nirhutnik) [reply docs/source/checks/vision/train_test_validation/plot_property_label_correlation_change.py:3]: Fixed

**Review by noamzbr — APPROVED** [review]
